pkg/rkecerts/expiration.go

```diff
@@ -0,0 +1,56 @@
+package rkecerts
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	rkeCluster "github.com/rancher/rke/cluster"
+	"github.com/rancher/rke/pki"
+	"github.com/rancher/rke/pki/cert"
+	v3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	v1 "k8s.io/api/core/v1"
+)
+
+func CleanCertificateBundle(certs map[string]pki.CertificatePKI) {
+	for name := range certs {
+		if strings.Contains(name, "client") ||
+			strings.Contains(name, "token") ||
+			strings.Contains(name, "header") ||
+			strings.Contains(name, "admin") {
+			delete(certs, name)
+		}
+	}
+}
+
+func GetCertExpiration(c string) (v3.CertExpiration, error) {
+	date, err := GetCertExpirationDate(c)
+	if err != nil {
+		return v3.CertExpiration{}, err
+	}
+	return v3.CertExpiration{
+		ExpirationDate: date.Format(time.RFC3339),
+	}, nil
+}
+
+func GetCertExpirationDate(c string) (*time.Time, error) {
+	certs, err := cert.ParseCertsPEM([]byte(c))
+	if err != nil {
+		return nil, err
+	}
+	return &certs[0].NotAfter, nil
```

Comment:
Probably index check for safety

Reply:
he already checked if len == 0, this should be guaranteed shouldnt it? do we need to handle if len > 1?

Reply:
I fixed it after his comment, this is resolved.

Reply:
What about Luther's question? If len > 1 what do?

Reply:
Technically we could loop and check all of them, but theres no reason one pem would have multiple certs with diff expiration dates, rke requires [specific separated pems](https://rancher.com/docs/rke/latest/en/installation/certs/#using-custom-certificates). This code was pre-existing, I just moved it.